### PR TITLE
Amm v0.2

### PR DESCRIPTION
**Pull request by dgornjakovic**
[Tip-19](https://github.com/thales-markets/thales-improvement-proposals/blob/main/TIPs/TIP-19.md)

### CODE REVIEW

File: contracts/AMM/SafeBox.sol [hunk 1]
```diff
@@ -0,0 +1,17 @@
+pragma solidity ^0.5.16;
```
Comment (grujawork) [inline contracts/AMM/SafeBox.sol:1]: can be 0.8.0?

File: contracts/AMM/ThalesAMM.sol [hunk 2]
```diff
@@ -125,7 +128,7 @@ contract ThalesAMM is ProxyOwned, ProxyPausable, ProxyReentrancyGuard, Initializ
             if (basePrice <= MIN_SUPPORTED_PRICE || basePrice >= MAX_SUPPORTED_PRICE) {
                 return 0;
             }
-            uint sell_max_price = basePrice.mul(ONE.sub(max_spread)).div(ONE);
+            uint sell_max_price = basePrice.mul(ONE.sub(max_spread.sub(safeBoxImpact))).div(ONE);
```
Comment (grujawork) [inline contracts/AMM/ThalesAMM.sol:131]: safeBoxImpact is an amount, not a percentage? so because of that is sub(traction)

**Review by grujawork — APPROVED** [review]

File: contracts/AMM/SafeBox.sol (hunk 1 above)
Comment (dgornjakovic) [reply contracts/AMM/SafeBox.sol:1]: have to deprecate all synthetix dependencies first

File: contracts/AMM/ThalesAMM.sol (hunk 2 above)
Comment (dgornjakovic) [reply contracts/AMM/ThalesAMM.sol:131]: its a percentage, but at that place we calculate the percentage to add to base price

**Pull request merged**